examples/tokens/ERC1155ownable.vy

```diff
@@ -0,0 +1,356 @@
+    # @param previousOwner The current, soon to be previous owner of the contract
+    # @param newOwner The new owner of the contract
+    previouwOwner: address 
+    newOwner: address
+
+event TransferSingle:
+    # @dev Emits on transfer of a single token
+    # @param operator the address initiating the transfer. owner or authorized operator
+    # @param fromAddress the address sending the token
+    # @param to the address receiving the token
+    # @param id the token ID
+    # @param value the amount of tokens
+    operator:   indexed(address)
+    fromAddress: indexed(address)
+    to: indexed(address)
+    id: uint256
+    value: uint256
+
+event TransferBatch:
+    # @dev Emits on batch transfer of tokens. the ids array correspond with the values array by their position
+    # @param operator the address initiating the transfer. owner or authorized operator
+    # @param fromAddress the address sending the token
+    # @param to the address receiving the token
+    # @param ids list of token ids
+    # @param values list of amounts for each token 
+    operator: indexed(address) # indexed
+    fromAddress: indexed(address)
+    to: indexed(address)
+    ids: DynArray[uint256, BATCH_SIZE]
+    values: DynArray[uint256, BATCH_SIZE]
+
+event ApprovalForAll:
+    # @dev This emits when an operator is enabled or disabled for an owner. The operator can manage
+    # @dev all NFTs of the owner.
+    # @param account Owner of NFT.
+    # @param operator Address to which we are setting operator rights.
+    # @param approved Status of operator rights(true if operator rights are given and false if revoked).
+    account: indexed(address)
+    operator: indexed(address)
+    approved: bool
+
+event URI:
+    # @dev This emits when the URI gets changed
+    # @param value the new URI
+    # @param id the ID 
+    value: String[MAX_URI_LENGTH]
+    id: uint256
+
+
+############### interfaces ###############
+implements: ERC165
+
+interface IERC1155Receiver:
+    def onERC1155Received(
+       operator: address,
+       sender: address,
+       id: uint256,
+       amount: uint256,
+       data: Bytes[CALLBACK_NUMBYTES],
+   ) -> bytes32: payable
+    def onERC1155BatchReceived(
+        operator: address,
+        sender: address,
+        ids: DynArray[uint256, BATCH_SIZE],
+        amounts: DynArray[uint256, BATCH_SIZE],
+        data: Bytes[CALLBACK_NUMBYTES],
+    ) -> bytes4: payable
+
+interface IERC1155MetadataURI:
+    def uri(id: uint256) -> String[MAX_URI_LENGTH]: view
+
+############### functions ###############
+
+@external
+def __init__(name: String[128], symbol: String[16], uri: String[1024]):
+    # @dev contract initialization on deployment
+    # @dev will set name and symbol, interfaces, owner and URI
+    # @dev self.paused will default to false
+    # @param name the smart contract name
+    # @param symbol the smart contract symbol
+    # @param uri the new uri for the contract
```

Comment:
Can these all be changed to docstrings so that they produce valid natspec?
```suggestion
    """
    @dev contract initialization on deployment
         will set name and symbol, interfaces, owner and URI
         self.paused will default to false
    @param name the smart contract name
    @param symbol the smart contract symbol
    @param uri the new uri for the contract
    """
```

Reply:
Changed to docstring